File: cfgs/java_cfgs_foo/PERFECT_REVERSIBLE_STRING.java
Nodes
1;special;;;
2;statement;int i = 0;14;
3;statement;int j = fun_1() - 1;14;
4;conditional;foo_1 < foo_2;15;
5;conditional;fun_1(foo_2) != fun_2(foo_4);16;
6;statement;return false;16;
7;exit;;;
8;statement;i++;17;
9;statement;j--;18;
10;statement;return true;20;
Edges
0;2;
1;3;
2;4;
3;10;5;
4;6;8;
5;7;
6;
7;9;
8;4;
9;7;

File: cfgs/pythoncfgs/PERFECT_REVERSIBLE_STRING.py
Nodes
1;statement;foo_1 = 0;8;
2;statement;foo_1 = fun_1(foo_2) - 1;8;
3;conditional;foo_1 < foo_2;9;
4;statement;return true;14;
5;conditional;foo_1[foo_2] != foo_3[foo_4];10;
6;statement;foo_1 += 1;12;
7;statement;foo_1 -= 1;13;
8;statement;return false;11;
9;exit;;
Edges
0;2;
1;3;
2;4;5;
3;9;
4;6;8;
5;7;
6;3;
7;9;
8;
